File: Examples/test-suite/php/director_min_runme.php
<?php

require "tests.php";

// No new functions
check::functions(array());
// New classes
check::classes(array('Callback', 'Caller'));
// No new vars
check::globals(array());

class MinCallback extends Callback {
  function run() {
    return true;
  }
}

$callback = new MinCallback();
$caller = new Caller($callback);
check::is_a($caller->call(), true);
check::done();

File: Examples/test-suite/perl5/director_min_runme.pl
use strict;
use warnings;
use Test::More tests => 5;
BEGIN { use_ok 'director_min' }
require_ok 'director_min';

{
  package MinCallback;
  use base 'director_min::Callback';
  sub run { 1; }
}

my $callback = MinCallback->new();
isa_ok $callback, 'MinCallback';
my $caller = director_min::Caller->new($callback);
isa_ok $caller, 'director_min::Caller';
is $caller->call(), 1, 'return true';
